File: Java/2125.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int previous = 0, result = 0;
        for (String s : bank) {
            int count = 0;
            for (int i = 0; i < s.length(); i++) {
                if (s.charAt(i) == '1') {
                    count++;
                }
            }
            if (count > 0) {
                result += previous * count;
                previous = count;
            }
        }
        return result;
    }
}

File: Python/2125.py
from typing import List


class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        result = prev_lasers = 0

        for row in bank:
            laser_count = 0
            for c in row:
                if c == "1":
                    laser_count += 1
            if laser_count == 0:
                continue
            result += prev_lasers * laser_count
            prev_lasers = laser_count

        return result
